lib/server.ml

```diff
@@ -205,11 +238,23 @@ module Command = struct
                         "Error in connection handler: %s"
                         (Printexc.to_string exn)))
                 (fun socket addr ->
-                  Switch.run (fun sw -> connection_handler ~sw socket addr)))
+                  Switch.run (fun sw ->
+                      let id = Openssl.random_string 20 in
```

Comment:
Probably an increment id would be better, especially as I would expect random_string to be a slow operation

Reply:
Also could be a per domain id